### PR TITLE
MAT-6476: add observations and stratifications to relevant elements

### PR DESCRIPTION
**Pull request by chubert-sb**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-6476](https://jira.cms.gov/browse/MAT-6476)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java [hunk 1]
```diff
@@ -135,10 +139,34 @@ private Set<String> getUsedDefinitionsFromMeasure(Measure measure) {
                           usedDefinitions.add(population.getDefinition());
                         }
                       });
+              if (group.getMeasureObservations() != null
```
Comment (nmorasb) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java]: do we have access to CollectionUtils in this project? If so we can replace collection null/empty checks with just CollectionUtils.isEmpty or CollectionUtils.isNotEmpty

File: src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java [hunk 2]
```diff
@@ -135,10 +139,34 @@ private Set<String> getUsedDefinitionsFromMeasure(Measure measure) {
...
+                group
+                    .getMeasureObservations()
+                    .forEach(
+                        measureObservation -> {
+                          if (!measureObservation.getDefinition().isEmpty()) {
+                            usedDefinitions.add(measureObservation.getDefinition());
+                          }
+                        });
+              }
+              if (group.getStratifications() != null && !group.getStratifications().isEmpty()) {
```
Comment (nmorasb) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java]: same thing regarding CollectionUtils.isEmpty

File: src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java (hunk 1 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java]: switched it over

File: src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java (hunk 2 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java]: switched it over

**Review by nmorasb — APPROVED** [review]
